pkg/sync/syncer.go

```diff
@@ -467,6 +468,9 @@ func (s *syncer) Sync(ctx context.Context) error {
 			if s.skipEntitlementsAndGrants {
 				s.state.SetShouldSkipEntitlementsAndGrants()
 			}
+			if s.skipGrants {
+				s.state.SetShouldSkipGrants()
+			}
```

Comment:
s.state.SetShouldSkipGrants() should probably get checked around line 510, and if it's true we should only push SyncEntitlementsOp, not SyncGrantsOp.